src/js_sync.hpp

```diff
@@ -62,11 +64,18 @@ static std::string partition_value_bson_to_string(typename T::Context ctx, typen
         bson::Bson partition_bson;
         if (Value<T>::is_string(ctx, partition_value_value)) {
             std::string pv = Value<T>::validated_to_string(ctx, partition_value_value);
+            if (pv.length() == 0) {
+                throw std::runtime_error("partitionValue of type 'string' may not be an empty string.");
+            }
             partition_bson = bson::Bson(pv);
         }
         else if (Value<T>::is_number(ctx, partition_value_value)) {
             auto pv = Value<T>::validated_to_number(ctx, partition_value_value);
-            partition_bson = bson::Bson(static_cast<int64_t>(pv));
+            if (pv < 0.0 || pv > JS_MAX_SAFE_INTEGER || fmod(pv, 1.0) > 0.0) {
```

Comment:
why do we need to validate the partition value to be an integer.

Reply:
As far as I'm informed, Sync doesn't support other numbers.

Reply:
👍

Reply:
Seems like my information was wrong, will update...